internal/server/add_contact_test.go

```diff
@@ -0,0 +1,205 @@
+		Errors: validationErrors,
+	}
+
+	template := &mockTemplates{}
+
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/133", strings.NewReader(""))
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	var returnedError error
+
+	testHandler := mux.NewRouter()
+	testHandler.HandleFunc("/{id}", func(w http.ResponseWriter, r *http.Request) {
+		returnedError = renderTemplateForAddContact(client, template)(sirius.DeputyDetails{}, w, r)
+	})
+
+	testHandler.ServeHTTP(w, r)
+
+	expectedValidationErrors := sirius.ValidationErrors{
+		"contactName": {
+			"isEmpty": "Enter a name",
+		},
+		"email": {
+			"isEmpty": "Enter an email address",
+		},
+		"phoneNumber": {
+			"isEmpty": "Enter a telephone number",
+		},
+		"isMainContact": {
+			"isEmpty": "Select whether this contact is a main contact",
+		},
+		"isNamedDeputy": {
+			"isEmpty": "Select whether this contact is the named deputy",
+		},
+	}
+
+	assert.Equal(addContactVars{
+		Path:   "/133",
+		Errors: expectedValidationErrors,
+	}, template.lastVars)
+
+	assert.Nil(returnedError)
+}
+
+func TestAddContactFormatValidationErrors(t *testing.T) {
+	assert := assert.New(t)
+	client := &mockContactInformation{}
+
+	validationErrors := sirius.ValidationErrors{
+		"contactName": {
+			"stringLengthTooLong": "The name must be 255 characters or fewer",
+		},
+		"jobTitle": {
+			"stringLengthTooLong": "The job title must be 255 characters or fewer",
+		},
+		"email": {
+			"stringLengthTooLong": "The email must be 255 characters or fewer",
+		},
+		"phoneNumber": {
+			"stringLengthTooLong": "The telephone number must be 255 characters or fewer",
+		},
+		"otherPhoneNumber": {
+			"stringLengthTooLong": "The other telephone number must be 255 characters or fewer",
+		},
+		"contactNotes": {
+			"stringLengthTooLong": "The note must be 255 characters or fewer",
+		},
+	}
+
+	client.err = sirius.ValidationError{
+		Errors: validationErrors,
+	}
+
+	template := &mockTemplates{}
+
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/133", strings.NewReader(""))
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	var returnedError error
+
+	testHandler := mux.NewRouter()
+	testHandler.HandleFunc("/{id}", func(w http.ResponseWriter, r *http.Request) {
+		returnedError = renderTemplateForAddContact(client, template)(sirius.DeputyDetails{}, w, r)
+	})
+
+	expectedValidationErrors := sirius.ValidationErrors{
+		"contactName": {
+			"stringLengthTooLong": "The name must be 255 characters or fewer",
+		},
+		"jobTitle": {
+			"stringLengthTooLong": "The job title must be 255 characters or fewer",
+		},
+		"email": {
+			"stringLengthTooLong": "The email must be 255 characters or fewer",
+		},
+		"phoneNumber": {
+			"stringLengthTooLong": "The telephone number must be 255 characters or fewer",
+		},
+		"otherPhoneNumber": {
+			"stringLengthTooLong": "The other telephone number must be 255 characters or fewer",
+		},
+		"contactNotes": {
```

Comment:
Would this test work just as well with only one error? We're just checking a Sirius error can be returned right?